gadget/mountedfilesystem.go

```diff
@@ -375,7 +379,27 @@ func (f *mountedFilesystemUpdater) Update() error {
 		}
 	}
 
-	if skipped == len(f.ps.ResolvedContent) {
+	deleted := false
+	if f.fromPs != nil {
+		for _, c := range f.fromPs.VolumeStructure.Content {
+			if knownContent[c.Target] {
+				continue
+			}
+			destPath, _ := f.entryDestPaths(f.mountPoint, "", c.Target, "")
+			if osutil.IsDirectory(destPath) {
+				continue
+			}
+			if err := os.Remove(destPath); err != nil {
```

Comment:
that'd be weird, but since we do not control what folks put in their gadget, some of the old files may have been added to the preserve list

Reply:
I did not see your review... It feels this one needs a bit of thinking. So I will do it when I back.

Reply:
Should not we instead verify before that all the files in `preserve` are also listed in the content? And if not, give a warning, and add it in the content for the rest to process.

Reply:
I wonder if there is a valid case where we want to keep the file, but not install a new one on new installations. In that case I understand. But `preserve` was not made for that.